server/src/com/cloud/tags/TaggedResourceManagerImpl.java

```diff
@@ -360,6 +348,53 @@ public void doInTransactionWithoutResult(TransactionStatus status) {
         return true;
     }
 
+    private List<? extends ResourceTag>  getResourceTags(final List<String> resourceIds, ResourceObjectType resourceType) {
+
+        List<String> uuids = new ArrayList<String>();
+        List<String> internalIds = new ArrayList<String>();
+        for(String resourceId : resourceIds){
+            if(resourceId.matches("^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$")){
```

Comment:
What about simplifying this ?
You can use `org.apache.commons.lang.math.NumberUtils.isDigits(String)`, to check if you have a long there, then you can convert it directly to Long, otherwise, you can check if it empty using `org.apache.commons.lang3.StringUtils.isNotBlank(CharSequence)`, and if it is not blank, you can add in the UUID list.

Reply:
yes. But validating uuid also is possibly more foolproof

Reply:
Well, yes and no. I was thinking about it yesterday, do we need to validate it really?
If it is a string, we could simply use it, if it is invalid, we will not find anything.
I see validation useful when you display a message back to the user saying, "hey, the data you entered is not following the expected format, which should be <formatExpected>...."
That is not what is happening here.